### ISSUE TITLE
Translation / Alignments

### ISSUE THREAD

**Issue by Jazzpirate** [KWARC/FoMID#18]
1. We might want a syntax for alignments in LaTeX
2. We might want users to be able to add new alignmentes on MathHub
3. => We probably need a better syntax and interfaces for users to specify/provide alignments including their arguments
4. MMT needs a better way to store/retrive/know about alignments => ULO, relational? (Work in progress, once relational is reimplemented as ULO/rdf)? Where/how should they be stored in the future to be persistent (if provided by a standalone user interface?)

**Comment by kohlhase** [comment]
>     1. We might want a syntax for alignments in LaTeX

we already have an alignment syntax in sTeX, it is in key/value arguments to `{module}` and `\symdef`.

**Comment by Jazzpirate** [comment]
> we already have an alignment syntax in sTeX, it is in key/value arguments to `{module}` and `\symdef`.

Right, but that would need to expanded, since this only carries the information "aligned with", not "aligned via" (e.g. argument orders, etc.)

**Comment by Jazzpirate** [comment]
In the most general case, we would need alignments to be able to map arbitrary terms according to patterns to arbitrary terms, i.e. something of the form "If `t` conforms to pattern `s(a1...an)`, translate to `s'(a1...an)`". That would subsume all computational translations I currently have. If we can get that to work in LaTeX and MMT, that is.
